exporters/otlp/connection.go

```diff
@@ -15,52 +15,96 @@
 package otlp
 
 import (
+	"context"
+	"fmt"
 	"math/rand"
+	"sync"
 	"sync/atomic"
 	"time"
 	"unsafe"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
 )
 
-func (e *Exporter) lastConnectError() error {
-	errPtr := (*error)(atomic.LoadPointer(&e.lastConnectErrPtr))
+type otlpConnection struct {
```

Comment:
All connection logic is contained in this object. The **otlpConnection** carries it's own configuration, and is responsible for dialing the collector, calling the **newConnectionHandler** callback and maintaining the connection alive.